new feature: removerange

SanjayDookhoo opened the pull request:
After creating your pull request, tick these boxes if they are applicable to you.

- [✓] I have tested my changes against the `dev` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [✓] I have tested my changes on Python 3.8 or higher
- [] I have ensured my code is formatted using [Black](https://github.com/psf/black)

----

### Description
Introduced a new feature to remove a specified range that is queued to play. Useful for when a entire playlist is queued, but you would like to remove a range of songs from playing.

I tried to take example of implementation from cmd_move (multiple values passed in) and the permissions from cmd_remove

### Related issues (if applicable)

SanjayDookhoo committed:
new feature: removerange

BabyBoySnow commented:
Hi there, like these changes, can you please ensure the code is formatted using black, 
also I think this would be better as an expansion to remove, as opposed to an another command. Thanks!

SanjayDookhoo commented:
sounds good, i can make those changes!

SanjayDookhoo committed:
recommended fixes:
- expanded remove cmd functionality with what removerange did, and removed removerange
- remove alias for removerange
- formatted my code using Black

SanjayDookhoo commented:
hey, I made the recommended changes, expanded on the cmd_remove  instead and formatted using Black

for the most part i just check to see if an extra position is used with the cmd_remove and if so, i apply the removal logic that was there for cmd_removerange, otherwise cmd_remove logic and how it worked was untouched. Also added the necessary informative changes to the help msg.

itsTheFae commented on musicbot/bot.py:
```diff
@@ -5292,10 +5292,16 @@ async def cmd_clear(
-        usage=["{cmd} [POSITION]"],
+        usage=[
+            "{cmd} <POSITION> <TO>\n"
+            + _Dd("    Remove song at POSITION.\n")
```
Each form of usage should be its own list entry here. 
```python
usage=[
    "{cmd} [POSITION]\n"
    + _Dd("    Remove a song at the end of the queue or at [POSITION].\n"),

    "{cmd} <FROM> <TO>\n"
    + _Dd("    Remove songs from position FROM to position TO.\n"),
],
```
The square brackets denote an optional argument, as remove previously allowed.

BabyBoySnow commented:
> hey, I made the recommended changes, expanded on the cmd_remove instead and formatted using Black
> 
> for the most part i just check to see if an extra position is used with the cmd_remove and if so, i apply the removal logic that was there for cmd_removerange, otherwise cmd_remove logic and how it worked was untouched. Also added the necessary informative changes to the help msg.

Awesome, thanks for doing that! I'll double check with a test tonight to make sure everythings working as intented and if nothing pops I'll I'll go ahead and merge this! Thanks for contributing! 🫡

itsTheFae commented on musicbot/bot.py:
```diff
@@ -5307,7 +5313,8 @@ async def cmd_remove(
         permissions: PermissionGroup,
         player: MusicPlayer,
-        index: str = "",
+        position: str,
```
The `position` parameter may need to have a default value to allow for the optional position argument.

SanjayDookhoo replied:
noted, i will also fix this, thanks for the feedback!

itsTheFae commented:
Couple things to address, but otherwise looks good to me.  
I'm also realizing the user-mention feature was never documented, or I missed it.  May want to add a third usage form such as:  
```python
"{cmd} <@USER>\n"
+ _Dd("    Remove songs added by the mentioned user.\n"),
 ```

SanjayDookhoo commented:
i can also look into removing with user mention, i would like to, im not sure if i should have a look at the user mention remove after this pull request is closed, so it can be a seperate pull request for a different enhancement

itsTheFae commented:
We don't want to remove the feature.  It's been a feature for a long time, I just failed to add documentation when everything got converted for translations and doc generation.

SanjayDookhoo commented:
ohhh i understand now, no problem, I will add that documentation with my next push, thanks for the clarification

SanjayDookhoo committed:
improved documentation and fixed the default value being set for cmd_remove

SanjayDookhoo commented:
thanks for the feedback! the changes were made with regards to the feedback

BabyBoySnow commented:
I have only two concerns after some quick "that shouldn't be used like that" but someone probably will lol.

https://prnt.sc/OWOMXmLcwtjQ

https://prnt.sc/uSLDYZmJreCP

SanjayDookhoo committed:
cmd_remove [POSITION] checks to ensure POSITION is a positive integer

SanjayDookhoo commented:
fair, i added in a small check while in it for cmd_remove [POSITION] to ensure POSITION is a positive integer
